### DISCUSSION TITLE
Dependency version upgrades? (react-keyed-flatten-children)

### DISCUSSION THREAD

**Discussion by atoko** [cloudscape-design/components#3080] [General]
Hi team. I've been having some issues when bundling cloudscape because of it's dependency on `react-keyed-flatten-children`. The version that is currently in use doesn't play well with ESM/Cjs (1.3.0).

This was fixed sometime back, and newer react versions have bumped their dependency to at least version 3. Is there any plan for bringing this up to date? 

Offtopic: It is such a small utility that sometimes I wonder if it's worth inlining into the package and attributing credit appropriately. At this point it's no longer cache friendly with React (main reason to split it out, in theory you would only have to download the dep once), but as it is I now have 2 versions in node_modules

**Comment by cansuaa** [comment]
Hi, thank you for bringing this to our attention. I will create a ticket on our end and update this discussion later on when I have more updates.

**Comment by jperals** [comment]
Hi @atoko,

I tried upgrading to 3.0.2 but it breaks some of our consumers, who don't have support for native ESM in their Jest environments. Version 2.2.1 however works, and since it contains [this commit](https://github.com/grrowl/react-keyed-flatten-children/commit/c14a0d9eeb2b0b5d5ec852a0642d717c5b80d9f7) I believe it should work for you as well. Could you confirm?

**Comment by jperals** [comment]
The dependency has been upgraded to version 2.2.1, and I will now close this discussion. Feel free to open a new one if you are still facing issues.
